playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml

```diff
@@ -173,7 +173,9 @@
   - name: Reconcile Cluster Roles
     command: >
       {{ openshift.common.client_binary }} adm --config={{ openshift.common.config_base }}/master/admin.kubeconfig
-      policy reconcile-cluster-roles --additive-only=true --confirm
+      policy reconcile-cluster-roles --additive-only=true --confirm -o name
+    register: reconcile_cluster_role_result
+    changed_when: reconcile_cluster_role_result.stdout.length > 0
```

Comment:
We want to also check the rc for the `changed_when` because `-o name` will tell us what objects need a change, not whether or not they did change (per IRC discussion with liggit).

I would agree that we would also want to have a `failed_when` that examined the `rc != 0`